azext_iot/tests/test_iot_central_int.py

```diff
@@ -229,6 +229,19 @@ def test_central_device_registration_info_unassociated(self):
             == "Device does not have a valid template associated with it."
         )
 
+    def test_central_device_registration_summary(self):
+
+        result = self.cmd(
+            "iot central app device registration-summary --app-id {}".format(APP_ID,)
+        )
+
+        json_result = result.get_output_in_json()
+        assert json_result[DeviceStatus.provisioned.value] is not None
+        assert json_result[DeviceStatus.registered.value] is not None
+        assert json_result[DeviceStatus.unassociated.value] is not None
+        assert json_result[DeviceStatus.blocked.value] is not None
+        assert len(json_result) == 4
```

Comment:
Nice!!

If we want to be flexible to DeviceStatus updates we can do this:

```
assert len(json_result) == len(DeviceStatus)
for val in json_result.values():
    assert val >= 0
```

This change means that if we add or remove fields from DeviceStatus that this test will still work.

If we want to enforce that only these 4 props should ever be present then keep the test as is

Reply:
It will give us the additional benefit of checking that the values in the dictionary are `int`s

---
In reply to: [428987160](https://github.com/Azure/azure-iot-cli-extension/pull/192#discussion_r428987160) [](ancestors = 428987160)

Reply:
#wontfix - I would prefer to keep it this way so that the test fails for any change.